### PR TITLE
[flutter_tools] remove VmService screenshot for native devices.

### PR DESCRIPTION
**Pull request by jonahwilliams**
* This is completely broken on the Impeller renderer, see: https://github.com/flutter/flutter/issues/135052
* Even on the Skia renderer, this gives a software rasterized screenshot which will absolutely look different from a native rendering screenshot.

I plan to remove this functionality from the engine.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/terminal_handler_test.dart [hunk 1]
```diff
@@ -1018,35 +1018,11 @@ void main() {
   testWithoutContext('s, can take screenshot on debug device that does not support screenshot', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/terminal_handler_test.dart:1018]: what is this testing?

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/terminal_handler_test.dart:1018]: oh, was this testing some kind of fallback behavior?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/terminal_handler_test.dart:1018]: Yeah, this was testing the fallback behavior and I updated to test that nothing happens.

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/terminal_handler_test.dart:1018]: can you update the test title?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/terminal_handler_test.dart:1018]: oh, right!

File: packages/flutter_tools/lib/src/commands/screenshot.dart [hunk 2]
```diff
@@ -33,21 +32,20 @@ class ScreenshotCommand extends FlutterCommand {
...
       valueHelp: 'type',
       help: 'The type of screenshot to retrieve.',
-      allowed: const <String>[_kDeviceType, _kSkiaType, _kRasterizerType],
+      allowed: const <String>[_kDeviceType, _kSkiaType],
       allowedHelp: const <String, String>{
         _kDeviceType: "Delegate to the device's native screenshot capabilities. This "
                       'screenshots the entire screen currently being displayed (including content '
                       'not rendered by Flutter, like the device status bar).',
         _kSkiaType: 'Render the Flutter app as a Skia picture. Requires "--$_kVmServiceUrl".',
-        _kRasterizerType: 'Render the Flutter app using the rasterizer. Requires "--$_kVmServiceUrl."',
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/screenshot.dart:50]: So this PR is only removing the ability to do `flutter screenshot --type=rasterizer --vm-service-url=$URL`?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/screenshot.dart:50]: Correct, though the Skia picture type will also eventually become irrelevant as that is only used for SKP debugging

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by zanderso — APPROVED** [review]

**Pull request merged**